### DISCUSSION TITLE
how to fix link undefined references abseil?

### DISCUSSION THREAD

**Discussion by imradzi** [googleapis/google-cloud-cpp#12601] [Q&A] [answered]
Hi,

I'm using C++20 and having problem linking to oauth2: 

curl_rest_client.cc:(.text+0x379): undefined reference to `absl::lts_20230802::StrCat[abi:cxx11](absl::lts_20230802::AlphaNum const&, absl::lts_20230802::AlphaNum const&)'

how do i fix this? thanks.

**Comment by coryan** [comment, marked as answer]
> how do i fix this? 

It depends.  We need a few more details to help:

- Can you give us some general information about your environment?
   - Platform? (Linux, macOS, Windows)
   - Can you tell us something about the compiler you are using? (GCC, Clang, MSVC)
   - Can you tell us what version?
- What build system are you using to build `google-cloud-cpp`? (CMake, Bazel)
- How did you compile and install Abseil? (package manager? manually?)
- How did you compile and install `google-cloud-cpp`? (package manager? manually? did you use the /doc/packaging.md instructions?)
- What build system are you using to build your application?
- How do you link `google-cloud-cpp`?
   - Did you use a CMake target like `google-cloud-cpp::storage`?
   - Did you use a Bazel target?
   - Did you use pkg-config and something like `$(pkg-config -libs google_cloud_cpp_storage)`?
   - Did you manually create the list of `-l` flags?


Depending on your answers the solution might be:

- Use the targets such as `google-cloud-cpp::storage` for CMake or `//:storage` for Bazel instead of manually listing `-l` flags.
- With older versions of Abseil, you must make sure you use a consistent or at least compatible set of flags when compiling Abseil and downstream dependencies.
- There is a bug in the build scripts for `google-cloud-cpp`.

**Reply by coryan** [reply]
Did we answer your question?  Are you able to successfully link the library?

**Comment by lucas-0liveira** [comment]
Hi,

It seems like you're encountering an issue with linking to the absl::lts_20230802::StrCat function in your C++20 code. This might be related to a missing or incorrect library or header file.

Make sure you have included the necessary library or header file for absl::lts_20230802::StrCat. This might be an ABSL library or a specific header file.

**Reply by coryan** [reply]
This looks like a LLM-generated answer.  It adds no value over what can be found with a google search.  I think we will start treating such answers as spam.
